### PR TITLE
STORM-3697 add capacity metric

### PR DESCRIPTION
**Pull request by agresch**
## What is the purpose of the change

Allow users to track a capacity metric other than on UI

## How was the change tested

Ran topology with load and validated reported capacity was in line with UI.

### CODE REVIEW

File: storm-client/src/jvm/org/apache/storm/metrics2/TaskMetrics.java [hunk 1]
```diff
@@ -50,6 +52,13 @@ public TaskMetrics(WorkerTopologyContext context, String componentId, Integer ta
         this.samplingRate = ConfigUtils.samplingRate(topoConf);
     }
 
+    public void setCapacity(double capacity) {
```
Comment (bipinprasad) [inline storm-client/src/jvm/org/apache/storm/metrics2/TaskMetrics.java:55]: Does this method need to guard against multiple calls, by tracking previous capacity and adding only the incremental difference to gauge.addValue()  - so the total capacity is always the capacity in the latest call to setCapacity? 

Alternatively, respecting only first call and ignoring subsequent calls.

**Review by kishorvpatil — APPROVED** [review]
👍

**Review by kishorvpatil — APPROVED** [review]
👍

**Review by Ethanlm — APPROVED** [review]

File: storm-client/src/jvm/org/apache/storm/stats/BoltExecutorStats.java [hunk 2]
```diff
@@ -53,10 +58,53 @@ public void cleanupStats() {
...
+
+        // Calculate capacity:  This is really for the whole executor, but we will use the executor's first task
+        // for reporting the metric.
+        double capacity = calculateCapacity(workerUptimeSecs);
+        firstExecutorTask.getTaskMetrics().setCapacity(capacity);
+    }
+
+    private double calculateCapacity(long workerUptimeSecs) {
+        if (workerUptimeSecs > 0) {
+            Map<String, Double> execAvg = valueStat(this.getExecuteLatencies()).get(MultiCountStat.TEN_MIN_IN_SECONDS_STR);
```
Comment (Ethanlm) [inline storm-client/src/jvm/org/apache/storm/stats/BoltExecutorStats.java:75]: `TEN_MIN_IN_SECONDS_STR` can be a local variable.  But this is not important.

Comment (agresch) [reply storm-client/src/jvm/org/apache/storm/stats/BoltExecutorStats.java:75]: I was going to use MultiCountStat.TEN_MIN_IN_SECONDS_STR to replace the one in StatsUtil, which is in storm-server for more general usage, but didn't replace the other usages to keep a smaller review.

Comment (Ethanlm) [reply storm-client/src/jvm/org/apache/storm/stats/BoltExecutorStats.java:75]: There is another similar class `MultiLatencyStat`. It doesn't feel right to use `MultiCountStat.TEN_MIN_IN_SECONDS_STR` for `MultiLatencyStat`. But this is just personal preference and we can leave this to the future. I am okay with current change.

**Pull request merged**